src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/ActivitiesService.kt

```diff
@@ -36,22 +36,19 @@ class ActivitiesService(
   }
 
   fun createActivity(dto: CreateActivityRequest): CreateActivityResponse =
-
-    activityRepository.save(mapActivityModel(dto)).run {
-
-      payRates = mapRates(dto, this)
-
-      telemetryClient.trackEvent(
-        "activity-created",
-        mapOf(
-          "id" to courseActivityId.toString(),
-          "prisonId" to prison.id,
-        ),
-        null
-      )
-
-      CreateActivityResponse(courseActivityId)
-    }
+    mapActivityModel(dto)
+      .apply { payRates.addAll(mapRates(dto, this)) }
+      .also {
+        telemetryClient.trackEvent(
+          "activity-created",
+          mapOf(
+            "id" to it.courseActivityId.toString(),
+            "prisonId" to it.prison.id,
+          ),
+          null
+        )
+      }
+      .let { CreateActivityResponse(activityRepository.save(it).courseActivityId) }
```

Comment:
Switched repository save to the end as we can let Hibernate handle the pay rates.